src/main/java/frc/robot/subsystems/vision/Limelight.java

```diff
@@ -0,0 +1,92 @@
+package frc.robot.subsystems.vision;
+
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.geometry.Translation2d;
+import edu.wpi.first.net.PortForwarder;
+import edu.wpi.first.networktables.NetworkTable;
+import edu.wpi.first.networktables.NetworkTableEntry;
+import edu.wpi.first.networktables.NetworkTableInstance;
+import frc.robot.subsystems.LoggedSubsystem;
+
+import java.util.Optional;
+import java.util.OptionalDouble;
+
+public class Limelight extends LoggedSubsystem<LimelightLogInputs> {
```

Comment:
Could you please explain why is `LoggedSubsystem` required? I don't see 6328 needing it so I'd love to understand.
I suggest you take a look at [their vision code implementation from last year](https://github.com/Mechanical-Advantage/RobotCode2022/tree/main/src/main/java/frc/robot/subsystems/vision).

Reply:
I don't think it's required but Eitan said we should use it

Reply:
This is not a good explanation.
Even if Dean Kamen came and told you to use `LoggableSystem`, I want you to know exactly why it should be used.

Reply:
I think they are using logged subsystem for the limelight because they want to know what is the distance and the angle to the target. Also, it will help them later during the competition after a match to investigate problems

Reply:
First, use "we" — you are part of the team! 😄
Second, I still don't understand why `LoggedSubsystem` is required. AdvantageKit supports all the logging stuff by itself, it's one of its main features.
I recommend again looking at 6328's code linked above, it should be sufficient.